boot_services/src/boot_services.rs

```diff
@@ -1606,4 +1606,72 @@ mod test {
         let status = boot_services.free_pool(ptr::null_mut());
         assert_eq!(status, Err(efi::Status::INVALID_PARAMETER));
     }
+    
+    #[test]
+    fn test_locate_protocol() {
+        const DEVICE_PATH_PROTOCOL: protocol_handler::DevicePath = protocol_handler::DevicePath {};
+        use r_efi::protocols::device_path;
+        static DEVICE_PATH_PROTOCOL_INTERFACE: device_path::Protocol = unsafe { MaybeUninit::zeroed().assume_init() };
+        let boot_services = boot_services!(locate_protocol = efi_locate_protocol);
+
+        extern "efiapi" fn efi_locate_protocol(
+            protocol_guid: *mut efi::Guid,
+            registration: *mut c_void,
+            interface: *mut *mut c_void,
+        ) -> efi::Status {
+            unsafe {
+                assert!(!protocol_guid.is_null(), "Protocol guid should not be null");
+                assert!(registration.is_null(), "Registration should be a null pointer");
+                assert!(!interface.is_null(), "Interface should not be a null pointer");
+                assert_eq!(
+                    protocol_guid.as_mut().unwrap(),
+                    &device_path::PROTOCOL_GUID,
+                    "Protocol guid should have been Device Path guid"
+                );
+                let device_path_interface = interface as *mut *mut device_path::Protocol;
+
+                *device_path_interface =

```

Comment:
consider using the ptr.write() method instead of deref and assign:
```
interface.write(&DEVICE_PATH_PROTOCOL_INTEFACE as *const c_void as *mut c_void);
```

I've seen arguments that this pattern is better because it allows for error checking on the write (in particular, that interface is not null).